src/poetry/repositories/pool.py

```diff
@@ -136,8 +135,9 @@ def package(
             raise ValueError(f'Repository "{repository}" does not exist.')
 
         if repository is not None and not self._ignore_repository_names:
-            with suppress(PackageNotFound):
-                return self.repository(repository).package(name, version, extras=extras)
```

Comment:
Are you sure that `PackageNotFound` can't be raised? For example `PyPiRepository.package()` calls `get_release_info()`, which can raise `PackageNotFound`.

Reply:
Ah, good spot.  I had found the method on the base `Repository` class, where it does not raise the exception, but does return `None`.

That's inconsistent with all of the implementations on inherited classes, which do raise the exception and do not return None...

I'll change the base to raise the exception too, that seems the least invasive.

Having said which: in this particular bit of code there really is no point in suppressing the exception - because the effect will only be that it gets raised anyway ten lines later...  I'll tidy that too.